src/app/controllers/registerCoach.ts

```diff
@@ -0,0 +1,60 @@
+import express = require('express')
+import Coach from '../models/coachModel'
+import sendEmail from '../helpers/sendEmail'
+import coachValidator from '../helpers/validators'
+import * as BCrypt from 'bcrypt'
+
+async function sendFormInfo(req: express.Request, res: express.Response) {
+  
+    let aCoach = req.body
+
+    await BCrypt.hash(aCoach.password, 10, async (err, hash) => {
```

Comment:
Potentially generating a hash (which is actually time intensive) for an invalid entry here, I'd suggest validating before hand. A really nice solution to this in express is using validation middleware that can validate the request body (or whatever part of the request needs it) before it ever gets to the controller. You can then just import your middleware into routes and apply it to the specific routes you need.

Here's an example: https://github.com/ashleycoles/ac-laptimes/blob/master/Middleware/Users/UsersValidator.js

Reply:
You could also hash the password as part of this, validate and then hash, saving the hash over the original password in the req.body. Stops (sort of) any muppet from being able to accidentally start storing plain text passwords down the line